include/lsst/meas/modelfit/RegularizedMoments.h

```diff
@@ -0,0 +1,123 @@
+ *
+ * Developed for the LSST Data Management System.
+ * This product includes software developed by the LSST Project
+ * (https://www.lsst.org).
+ * See the COPYRIGHT file at the top-level directory of this distribution
+ * for details of code ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#ifndef LSST_MEAS_MODELFIT_REGULARIZEDMOMENTS_H
+#define LSST_MEAS_MODELFIT_REGULARIZEDMOMENTS_H
+
+#include <Eigen/Dense>
+
+namespace lsst {
+namespace meas {
+namespace modelfit {
+
+/**
+ *  @brief This class is a model for calculating observed shape moments and their derivatives
+ *
+ *  This class models the biased weighted moments of an astronomical source given a vector of
+ *  moments intrinsic to an object, and a vector of moments representing a weighting function.
+ *
+ *  The class is constructed with the moments of weight function, and the moments of the object
+ *  are set by calling the at method. This allows a single instance to be reused to evaluate
+ *  multiple positions in object moment space. This structure is computationally more efficient
+ *  as well since the weight function would remain constant for many evaluation of image moments 
+ *
+ *  This class is designed to be used as an objective function in an optimizer which will find the
+ *  maximally likely image moments given moments measured from an image. As such the class
+ *  provides two methods in addition to the at method already mentioned. One method computes the
+ *  biased weighted moments given the specified weight and "true" moments, and the other computes
+ *  the derivative of the objective function along each of the image moment variables.
+ */
+class MomentsModel {
+public:
+    using Moments = Eigen::Matrix<double, 6, 1>;
+    using Jacobian = Eigen::Matrix<double, 6, 6>;
+    using FirstMoment = Eigen::Matrix<double, 2, 1>;
+    using SecondMoment = Eigen::Matrix<double, 2, 2>;
+
+    /**
+     * Construct a MomentsModel object with a given vector of weight function moments
+     *
+     * @param[in]   W       An eigen vector of the moments of the elliptical Gaussian weighting function.
+     *                      Most likely the first element (the zeroth moment) should be one, representing
+     *                      a normalized weight function.
+     */
+    MomentsModel(Moments const & W): W(W) {
+    }
```

Comment:
Please also explicitly `default` or `delete` copy/move constructors / assignment operators and destructor.

Reply:
done